pkg/query/query_bench_test.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package query
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/thanos-io/thanos/pkg/gate"
+	"github.com/thanos-io/thanos/pkg/store/labelpb"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+	storetestutil "github.com/thanos-io/thanos/pkg/store/storepb/testutil"
+	"github.com/thanos-io/thanos/pkg/testutil"
+)
+
+// TestQuerySelect benchmarks querier Select method. Note that this is what PromQL is using, but PromQL might invoke
```

Comment:
Surely `TestQuerySelect` tests the querier select method, and `BenchmarkQuerySelect` benchmarks the querier select method?